docs/develop/_changelog.md

```diff
@@ -1,4 +1,23 @@
+# Changelog
+
+```{include} ../../CHANGELOG.md
+:end-before: <!-- towncrier release notes start -->
+:start-after: (DO-NOT-REMOVE-versioning-promise-START)
+
+```
+
+<!-- markdownlint-disable-next-line no-space-in-emphasis -->
+## {{ release_l }}, as of {sub-ref}`today` _{subscript}`/UNRELEASED DRAFT/`_
+
+```{important} This version is not yet released and is under active development
+```
+
+```{towncrier-draft-entries}
```

Comment:
This injects a pre-rendered changelog composed of the Towncrier fragments in the `docs/changelog-fragments.d/` before the release happens. As a part of the release preparation, you'd normally run a command that deletes those fragments from Git and adds the rendered section to `CHANGELOG.md`.